internal/promotion/runner/builtin/tar_extractor_test.go

```diff
@@ -0,0 +1,509 @@
+package builtin
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/pkg/promotion"
+	"github.com/akuity/kargo/pkg/x/promotion/runner/builtin"
+)
+
+func Test_tarExtractor_run(t *testing.T) {
```

Comment:
It would be nice to have a test case for files that are too big as well. I think you can test this doing something like:

```go
        // Mock large file
        largeFileSize := MaxDecompressedFileSize + 1
        hdr := &tar.Header{
            Name: "large_file.bin",
            Mode: 0600,
            Size: largeFileSize,
        }
        require.NoError(t, tw.WriteHeader(hdr))
        
        // Write zeros to the file
        _, err = tw.Write(make([]byte, largeFileSize))
        require.NoError(t, err)
```